gpiozero/pins/data.py

```
@@ -298,7 +489,144 @@ class PinInfo(namedtuple('PinInfo', (
+
+    .. attribute:: columns
+
+        The number of columns on the header.
+
+    .. attribute:: pins
+
+        A dictionary mapping physical pin numbers to :class:`PinInfo` tuples.
     """
+    def _func_style(self, function, style):
+        if function == V5:
+            return style('bold red')
+        elif function in (V3_3, V1_8):
+            return style('bold cyan')
+        elif function in (GND, NC):
+            return style('bold black')
+        elif function.startswith('GPIO') and function[4:].isdigit():
+            return style('bold green')
+        else:
+            return style('yellow')
+
+    def _format_full(self, style):
+        Cell = namedtuple('Cell', ('content', 'align', 'style'))
+
+        pin_digits = len(str(self.rows * self.columns))
+        lines = []
+        for row in range(self.rows):
+            line = []
+            for col in range(self.columns):
+                pin = (row * self.columns) + col + 1
+                try:
+                    pin = self.pins[pin]
+                    cells = [
+                        Cell(pin.function, '><'[col % 2], self._func_style(pin.function, style)),
+                        Cell('(%d)' % pin.number, '><'[col % 2], ''),
+                        ]
+                    if col % 2:
+                        cells = reversed(cells)
+                    line.extend(cells)
+                except KeyError:
+                    line.append('')
+            lines.append(line)
+        cols = list(zip(*lines))
+        col_lens = [max(len(cell.content) for cell in col) for col in cols]
+        lines = [
+            ' '.join(
+                '{cell.style}{cell.content:{cell.align}{width}s}{style:reset}'.format(
+                    cell=cell, width=width, style=style)
+                for cell, width, align in zip(line, col_lens, cycle('><')))
+            for line in lines
+            ]
+        return '\n'.join(lines)
+
+    def _format_row(self, row, style):
```

Should `_format_row` and `_format_col` raise an exception if passed an invalid row or column?
`print('{0:col4}'.format(pi_info('0002').headers['P1']))` _appears_ to work but obviously produces invalid output.